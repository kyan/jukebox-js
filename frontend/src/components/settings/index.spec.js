import React from 'react'
import { mount } from 'enzyme'
import GoogleAuthContext from '../../contexts/google'
import Settings from './index'

describe('Settinsg', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    describe('when authorised user', () => {
      const signOutMock = jest.fn()
      const mockGoogle = {
        signOut: signOutMock,
        isSignedIn: true,
        googleUser: {
          profileObj: {
            name: 'Fred Spanner',
            imageUrl: 'myImage123'
          }
        }
      }
      const wrapper = mount(
        <GoogleAuthContext.Provider value={mockGoogle}>
          <Settings />
        </GoogleAuthContext.Provider>
      )

      it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('handles signing out', () => {
        wrapper.find('Image').simulate('click')
        expect(signOutMock).toHaveBeenCalled()
      })
    })

    describe('when nonauthorised user', () => {
      const signInMock = jest.fn()
      const mockGoogle = {
        signIn: signInMock,
        isSignedIn: false,
        googleUser: null
      }
      const wrapper = mount(
        <GoogleAuthContext.Provider value={mockGoogle}>
          <Settings />
        </GoogleAuthContext.Provider>
      )

      it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('handles signing in', () => {
        wrapper.find('Button').simulate('click')
        expect(signInMock).toHaveBeenCalled()
      })
    })
  })
})
