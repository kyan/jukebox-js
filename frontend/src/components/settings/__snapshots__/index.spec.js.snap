// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settinsg render when authorised user renders as expected 1`] = `
<Settings>
  <Image
    as="img"
    floated="right"
    onClick={[Function]}
    rounded={true}
    size="mini"
    src="myImage123"
    title="Fred Spanner"
    ui={true}
  >
    <img
      className="ui mini rounded right floated image"
      onClick={[Function]}
      src="myImage123"
      title="Fred Spanner"
    />
  </Image>
</Settings>
`;

exports[`Settinsg render when nonauthorised user renders as expected 1`] = `
<Settings>
  <Button
    as="button"
    className="jb-settings-toggle"
    floated="right"
    icon="power off"
    onClick={[Function]}
    title="Login using Google"
  >
    <button
      className="ui icon right floated button jb-settings-toggle"
      onClick={[Function]}
      title="Login using Google"
    >
      <Icon
        as="i"
        name="power off"
      >
        <i
          aria-hidden="true"
          className="power off icon"
        />
      </Icon>
    </button>
  </Button>
</Settings>
`;
